### PR TITLE
[MM-45052] Add error checking in insights API

### PR DESCRIPTION
**Pull request by cyrilzhang-mm**
#### Summary

Change some function calls in accordance with the changes in https://github.com/mattermost/mattermost-server/pull/21048

#### Ticket Link

[Jira ticket](https://mattermost.atlassian.net/browse/MM-45052)

### CODE REVIEW

**Review by plant99 — APPROVED** [review]

File: server/api/insights.go [hunk 1]
```diff
@@ -113,7 +113,12 @@ func (a *API) handleTeamBoardsInsights(w http.ResponseWriter, r *http.Request) {
 		userLocation = time.Now().UTC().Location()
 	}
 	// get unix time for duration
-	startTime := mmModel.StartOfDayForTimeRange(timeRange, userLocation)
+	startTime, appErr := mmModel.StartOfDayForTimeRange(timeRange, userLocation)
+	if appErr != nil {
+		a.errorResponse(w, r, appErr)
```
Comment (plant99) [inline server/api/insights.go:118]: Can we create a new error with `ErrBadRequest` with just message from `appErr`? Same for the changes in user boards insights?

**Review by plant99 — CHANGES_REQUESTED** [review]

**Review by BenCookie95 — APPROVED** [review]
Nice work!

**Review by sbishel — APPROVED** [review]

**Review by plant99 — APPROVED** [review]

**Comment by sbishel** [comment]
Is this ready to merge?

**Comment by cyrilzhang-mm** [comment]
> Is this ready to merge?

No, the corresponding PR has not been merged in mattermost-server yet. We need to merge the PRs together, as discussed here:
https://github.com/mattermost/mattermost-server/pull/21048#issuecomment-1262492850

**Review by BenCookie95 — APPROVED** [review]

**Review by plant99 — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]

**Pull request merged**